internal/support/scheduler/operators/intervalaction/get.go

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright 2019 VMware Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package intervalaction
+
+import (
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type CollectionExecutor interface {
+	Execute() ([]contract.IntervalAction, error)
+}
+
+type intervalActionLoadAll struct {
+	database IntervalActionLoader
+	limit    int
+}
+
+func (op intervalActionLoadAll) Execute() ([]contract.IntervalAction, error) {
+	intervalActions, err := getIntervalActions(op.database, op.limit)
+	if err != nil {
+		return intervalActions, err
+	}
+	return intervalActions, nil
+}
+
+func NewAllExecutor(db IntervalActionLoader, limit int) CollectionExecutor {
+	return intervalActionLoadAll{
+		database: db,
+		limit:    limit,
+	}
+}
+
+func getIntervalActions(db IntervalActionLoader, limit int) ([]contract.IntervalAction, error) {
+	var err error
+	var intervalActions []contract.IntervalAction
+
+	if limit <= 0 {
+		intervalActions, err = db.IntervalActions()
```

Comment:
I think we are missing some additional logic here. We need to also ensure the number of results do not exceed the `Configuration.Service.MaxResultCount` value. With that being said I think we should always pass a limit to the `db.IntervalActionsWithLimit(limit)`. If my assumption is correct then to add to my previous comment regarding the simplification and removal of the function is further supported as this function would only call `db.IntervalActionsWithLimit(limit)` with either the specified `limit` or `Configuration.Service.MaxResultCount`.

Reply:
That's true. However, I think this should be done in the http handler as doing the check directly here will introduce a cyclic dependency. It is possible to pass it as part of the struct and check inside. Up to you for final implementation.

Reply:
I think passing in a copy of the configuration to the operator will not create a cycle in dependencies similar to what we are doing here:

https://github.com/edgexfoundry/edgex-go/blob/57817b912cbc23d07f161b86d5b5397305a543ba/internal/core/data/operators/value_descriptor/get.go#L33-L38 

and using that to check the `MaxResultCount` like:

https://github.com/edgexfoundry/edgex-go/blob/57817b912cbc23d07f161b86d5b5397305a543ba/internal/core/data/operators/value_descriptor/get.go#L47-L49

rather than referencing the global configuration variable.

Reply:
This endpoint doesn't have a limit GET parameter so by using db.IntervalActionsWithLimit(limit), limit would actually be Configuration.Service.MaxResultCount that is passed from the http handler to the operator's Execute. In that case, how will we even get to a point where we have LimitExceededException as we are the ones specifying how many records to fetch?

I think we should either do db.IntervalActions() and do the check or just do db.IntervalActionsWithLimit(limit) where a check wouldn't be required. The former is going to be slower though.

Reply:
Good point! I agree, this can go either way. Even though I like the idea of using a limit for performance reasons, I think it may be better to have the `intervalActionLoadAll` have a field for the `config.ServiceInfo` and load all the entities with ` db.IntervalActions()`. That way we can check the number of entries against the `MaxResultCount` and decide to either return the results or an `ErrLimitExceeded`. Doing it the other way, which will perform better with large amount of database entries, the user will lose information since they would only get a subset of the data they are trying to obtain, and they will have no indication that the data they have been given is only a subset. The also makes it more difficult to suggest since a `GET` to an endpoint with no limit, range, or filter criteria should return all the results or an error, I do not think there is any decent middle ground there from an API usage point-of-view. I hope this rant makes sense.